src/main/java/com/redhat/devtools/lsp4ij/client/features/EditorBehaviorFeature.java

```diff
@@ -86,6 +86,19 @@ public boolean isEnableEnterBetweenBracesFix(@NotNull PsiFile file) {
         return false;
     }
 
+    /**
+     * Whether or not editor improvements for nested braces/brackets/parentheses in TextMate files are enabled.
+     * Defaults to false.
+     *
+     * @param file the file
+     * @return true if editor improvements for nested braces/brackets/parentheses in TextMate files are enabled;
+     * otherwise false
+     */
+    public boolean isEnableTextMateNestedBracesImprovements(@NotNull PsiFile file) {
```

Comment:
The new editor behavior feature flag in the `EditorBehaviorFeature`. Disabled by default for custom language server definitions, but is overridden below to be enabled by default for user-defined language defintions where it can be disabled easily via client config.